### PR TITLE
Pass `{USER, COREOS}_DATA` separately in `testcloud`

### PR DESCRIPTION
**Pull request by frantisekz**
Together with now staged testcloud-0.11.0, should resolve https://github.com/teemtee/tmt/issues/3235 once and for all.

Required testcloud is built at: https://copr.fedorainfracloud.org/coprs/frantisekz/testcloud-temp/

The {USER, COREOS}_DATA variables were affected by threads overwriting them directly in testcloud's config object as observed in https://github.com/teemtee/tmt/pull/3244#issuecomment-2392236256 and following comments. Testcloud-side change can be seen here: https://pagure.io/testcloud/c/ee1202bcf57cd475f4586f3693e2127f9a12e92e?branch=master . 

Only these two variables can differ between various instances in tmt's use of testcloud, the others are kept separately since the domain api switchover last year: https://github.com/teemtee/tmt/pull/2078 .

Co-authored by @jskladan 

@happz @psss 

Pull Request Checklist

* [x] implement the feature
* [x] include a release note

### CODE REVIEW

**Comment by frantisekz** [comment]
So, the CoreOS tests failed previously, the upcoming run has https://pagure.io/testcloud/pull-request/204 included which should resolve that too.

tldr. previously, we could use sock.connect_ex(("127.0.0.1", self.get_instance_port())) in testcloud CoreOS/user sessions to verify that ssh port is open, but something had changed in how this works in libvirt/qemu, and the call now instantly returns OK.

**Comment by frantisekz** [comment]
@psss Reading through the results, if I am not missing anything, the issues are mainly:

`fail: Command 'rpm -q --whatprovides tree-but-spelled-wrong diffutils || dnf install -y  tree-but-spelled-wrong diffutils' returned 1.`

which wouldn't seem caused by testcloud/this PR, and fedora-coreos failures, which I have problems reproducing locally. All in all, this seems to resolve the mysterious failures at least for normal Fedora Cloud :/

File: .packit.yaml [hunk 1]
```diff
@@ -150,19 +153,8 @@ jobs:
...
-    allowed_pr_authors: [ "packit", "psss", "lzachar" ]
-    allowed_committers: [ "packit", "psss", "lzachar" ]
-    dist_git_branches: *all-targets
-
   - job: bodhi_update
     trigger: commit
     dist_git_branches:
       - fedora-branched
-      - epel-9
+      - epel-9
```
Comment (happz) [inline .packit.yaml:160]:
```suggestion
      - epel-9

```

**Comment by psss** [comment]
Bumped the require again in 8f4d1db3 to get the regression fix.

**Comment by psss** [comment]
Fixed timeout and adjusted the test to exercise the custom command path only: 06d5618f
Filed a new issue to improve the timeout handling: #3305 
Included a short release note in: 31e9b73d

**Review by psss — APPROVED** [review]
Now seems to be working nice! :tada:

**Comment by psss** [comment]
@happz, and it's green! \o/

**Review by happz — APPROVED** [review]
